server/lib/routes.js

```diff
@@ -40,15 +42,28 @@ module.exports = function (config, i18n) {
     );
   }
 
+  function addVersion(unversionedUrl) {
+    return VERSION_PREFIX + unversionedUrl;
+  }
+
+  function removeVersion(versionedUrl) {
+    return versionedUrl.replace(VERSION_PREFIX_REGEXP, '');
+  }
+
   return function (app) {
     // handle password reset links
-    app.get('/v1/complete_reset_password', function (req, res) {
-      res.redirect(req.originalUrl.slice(3));
+    app.get(addVersion('/complete_reset_password'), function (req, res) {
+      res.redirect(removeVersion(req.originalUrl));
+    });
+
+    // handle account unlock links
+    app.get(addVersion('/complete_unlock_account'), function (req, res) {
+      res.redirect(removeVersion(req.originalUrl));
     });
 
     // handle email verification links
-    app.get('/v1/verify_email', function (req, res) {
-      res.redirect(req.originalUrl.slice(3));
+    app.get(addVersion('/verify_email'), function (req, res) {
+      res.redirect(removeVersion(req.originalUrl));
```

Comment:
@rfk - what do you think about this approach, using helper functions to add/rmove the version prefix?